drivers/flash/flash_stm32f4x.c

```diff
@@ -292,8 +292,7 @@ int flash_stm32_update_rdp(const struct device *dev, bool enable,
 	switch (current_level) {
 	case FLASH_STM32_RDP2:
 		if (!enable || !permanent) {
-			__ASSERT(false, "RDP level 2 is permanent and can't be "
-					"changed!");
+			LOG_ERR("RDP level 2 is permanent and can't be changed!");
 			return -ENOTSUP;
```

Comment:
I won't block on this but ideally, all changes non strictly related to the new functionality and impacting existing code (ie which can cause a regression) should be made on a different commit. This should ease review and allow selected revert in case of regression.